src/components/application_manager/include/application_manager/application_impl.h

```diff
@@ -577,8 +577,8 @@ class ApplicationImpl : public virtual Application,
   bool audio_streaming_allowed_;
   bool video_streaming_suspended_;
   bool audio_streaming_suspended_;
-  sync_primitives::Lock video_streaming_suspended_lock_;
-  sync_primitives::Lock audio_streaming_suspended_lock_;
+  bool video_streaming_stopped_;
+  bool audio_streaming_stopped_;
```

Comment:
As variant it may be easier to handle streaming operations if create a separate structure for this functionality
```
struct StreamingProcessor {
  bool approved_;
  bool allowed_;
  bool uspended_;
  bool stopped_;

  void StartStreaming(){...}
  void StopStreaming(){...}
  etc
  ...
};

StreamingProcessor audioStreaming_;
StreamingProcessor videoStreaming_;
```
But this could entail extensive refactoring of the ApplicationImpl class.